src/workerd/io/trace.c++

```diff
@@ -494,7 +494,8 @@ Tracer::Span Tracer::makeSpan(kj::StringPtr operationName,
   }
   kj::TimePoint durationStartTime = kj::origin<kj::TimePoint>();
   if (parentSpanContext != nullptr || overrideParent != nullptr) {
-    durationStartTime = kj::systemPreciseMonotonicClock().now();
+    auto delta = kj::systemPreciseCalendarClock().now() - overrideStartTime;
+    durationStartTime = kj::systemPreciseMonotonicClock().now() - delta;
```

Comment:
I think it'd be good to have an explanatory comment and/or a TODO to require both start time overrides in the future.